api/v1/controllers/samples.js

```diff
@@ -197,7 +196,8 @@ module.exports = {
         u.checkDuplicateRLinks(rLinks);
       }
 
-      redisModelSample.patchSample(req.swagger.params, req.user.name)
+      const userName = req.user ? req.user.name : undefined;
+      redisModelSample.patchSample(req.swagger.params, userName)
```

Comment:
What happens if the user is undefined? Does patchSample throw an error?
Shouldn't that be impossible since we're always enforcing tokens?

Reply:
I had it has req.user.name in my previous PR but later realized that if at all we were to access any of these endpoints using a collector, req,user will not be defined.